discovery/pkg/openstack/httplogger.go

```diff
@@ -56,7 +56,7 @@ func (lrt *LogRoundTripper) RoundTrip(request *http.Request) (*http.Response, er
 
 	if response.StatusCode == http.StatusUnauthorized {
 		if lrt.numReauthAttempts == 3 {
-			return response, fmt.Errorf("tried to re-authenticate 3 times with no success")
+			log.Fatal("tried to re-authenticate 3 times with no success")
```

Comment:
I am not sure we should be calling os.Exit deep within a library. I'd rather see us handling this in main.go or somewhere higher in the call stack. However, getting this bubbled up to main might be complicated with our current design, as the reconciler doesn't return an error. 

With all this said, wouldn't this be caught by the healthz + liveness probe?